pkg/api/customization/cluster/validator.go

```diff
@@ -49,3 +68,93 @@ func (v *Validator) validateLocalClusterAuthEndpoint(request *types.APIContext,
 
 	return nil
 }
+
+func (v *Validator) validateEnforcement(request *types.APIContext, data map[string]interface{}) error {
+
+	var spec mgmtclient.Cluster
+	if err := convert.ToObj(data, &spec); err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, "Cluster spec conversion error")
+	}
+
+	if !v.checkClusterForEnforcement(&spec) {
+		return nil
+	}
+
+	ma := gaccess.MemberAccess{
+		Users:     v.Users,
+		GrLister:  v.GrLister,
+		GrbLister: v.GrbLister,
+	}
+
+	//if user is admin, no checks needed
+	callerID := request.Request.Header.Get(gaccess.ImpersonateUserHeader)
+
+	isAdmin, err := ma.IsAdmin(callerID)
+	if err != nil {
+		return err
+	}
+	if isAdmin {
+		return nil
+	}
+
+	if !strings.EqualFold(settings.ClusterTemplateEnforcement.Get(), "true") {
```

Comment:
do you think we can make this check before the caller id? So we save on IsAdmin call :)